usb.c

```diff
@@ -1120,8 +1121,12 @@ int usb_context_scan(struct iio_scan_backend_context *ctx,
 		unsigned int interface = 0;
 
 		ret = libusb_open(dev, &hdl);
-		if (ret)
+		if (ret) {
+			if (ret != LIBUSB_ERROR_ACCESS)
+				perms = true;
 			continue;
+		}
+		perms = true;
```

Comment:
Sort of - if no devices are found - and you can open at least one - perms = true (which is the case when run as root), so it does differentiate between no devices, and no IIO devices.

If you can open one different device (not as root), then no IIO devices, and the wrong udev rule will be the same.